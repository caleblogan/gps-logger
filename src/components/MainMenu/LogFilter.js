import React from 'react';
import PropTypes from 'prop-types';

import {Menu} from 'semantic-ui-react';

import {Input} from 'semantic-ui-react';

const LogFilter = ({onChange, value}) => {
  return (
    <Menu.Item>
      <Input
        inverted transparent className='icon'
        icon='search' placeholder='Search Logs...'
        onChange={onChange}
        value={value}
      />
    </Menu.Item>
  );
};

LogFilter.propTypes = {
  onChange: PropTypes.func,
  value: PropTypes.string,
};

export default LogFilter;
