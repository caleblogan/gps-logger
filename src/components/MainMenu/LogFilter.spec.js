import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

import LogFilter from './LogFilter';

describe('LogFilter', () => {
  it('basic snapshot', () => {
    const tree = renderer
      .create(<LogFilter/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders input component', () => {
    const logfilter  = shallow(<LogFilter value={''} onChange={()=>''}/>);
    expect(logfilter.find('Input').length).toBe(1)
  })
  it('passes value to Input', () => {
    const logfilter = shallow(<LogFilter value='boo' onChange={()=>''}/>);
    expect(logfilter.find('Input').props()).toHaveProperty('value', 'boo')
  })
  it('onChange is passed to Input', () => {
    const fn = () => ''
    const logfilter = shallow(<LogFilter value='boo' onChange={fn}/>);
    expect(logfilter.find('Input').props()).toHaveProperty('onChange', fn)
  })
})
