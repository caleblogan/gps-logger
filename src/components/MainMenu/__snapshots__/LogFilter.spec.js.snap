// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogFilter basic snapshot 1`] = `
<div
  className="ui inverted transparent icon input icon"
>
  <input
    disabled={undefined}
    onChange={[Function]}
    placeholder="Search Logs..."
    tabIndex={undefined}
    type="text"
    value={undefined}
  />
  <i
    aria-hidden="true"
    className="search icon"
  />
</div>
`;
